### DISCUSSION TITLE
Why does the gallery extension show images from different paths?

### DISCUSSION THREAD

**Discussion by IngoBelka** [datenstrom/community#804] [Ask a question] [answered]
Hejsan,

I have some pictures in two different pathes
```
/media/images/rettungsboot
/media/images/rettungsboot-malbuch
```
I set a gallery statement on a page as follows
```
[gallery rettungsboot/*.*jpg - photoswipe 18%]
```
What I get, are alle pictures of both pathes mentioned above, but I only want to see those from the first path.

Is this an intended behaviour or do we have a little bug there?

Thank you for your efforts.

**Comment by IngoBelka** [comment]
Hej,

what I found out now is, that
```
[gallery rettungsboot/.*jpg - photoswipe 18%]
```
solves the problem.

But is this a logically solution?

**Comment by markseuffert** [comment, marked as answer]
Here's what your [regex](https://en.wikipedia.org/wiki/Regular_expression) says:

```
rettungsboot/*.*jpg
└──────────┘└┘└┘└─┘
           1 2 3  4
```
1.  text starting with `rettungsboot`
2. followed by zero or more slashes
3. followed by zero or more arbitrary characters
4. followed by `jpg`

If this is not what you want, the [gallery extension](https://github.com/annaesvensson/yellow-gallery) has some examples.

**Reply by IngoBelka** [reply]
I didn't read the Gallery manual carefully enough - sorry.

So, I've overseen, that it is a regex. Now the behaviour is clear.  I only thought, the astericks are wildcards, but in a regex they have a totally different meaning.

Thank you for pointing me on this. :-)

**Reply by markseuffert** [reply]
Happy to help, regex are fun. 😁

**Reply by IngoBelka** [reply]
> ... regex are fun. 😁

Yes they are, indeed! :thumbsup:
